.gitlab-ci.yml

```diff
@@ -33,18 +34,36 @@ variables:
 
 stages:
 
+  - 'DebOps roles (pre)'
   - 'DebOps roles (no deps)'
   - 'DebOps roles (1st deps)'
   - 'DebOps roles (2nd deps)'
   - 'DebOps roles (3rd deps)'
   - 'DebOps roles (4th deps)'
-  - 'DebOps roles (5th deps)'
-  - 'DebOps roles (6th deps)'
 
 
-# === Role testing templates === [[[1
+# === Workflow === [[[1
 
-.tpl_test_role_base: &test_role_base
+# Run a single pipeline when duplicate cases are possible
+# https://docs.gitlab.com/ee/ci/yaml/README.html#switch-between-branch-pipelines-and-merge-request-pipelines
+workflow:
+  rules:
+    - if: '$CI_COMMIT_BRANCH && $CI_OPEN_MERGE_REQUESTS && $CI_PIPELINE_SOURCE == "push"'
+      when: never
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+      when: always
+    - if: '$CI_COMMIT_BRANCH && $CI_OPEN_MERGE_REQUESTS'
+      when: never
+    - if: '$CI_COMMIT_BRANCH'
+      when: always
+    - if: '$CI_REF_PROTECTED'
```

Comment:
This is to have a full CI on branches such as master, or stable-x-y. Also, when in GitLab, you can set variable to "1" when running the pipeline manually to have all the jobs.